packages/headless/src/features/commerce/facets/facet-search-set/commerce-facet-search-actions.ts

```diff
@@ -32,12 +38,21 @@ const getExecuteFacetSearchThunkPayloadCreator =
   > =>
   async (facetId: string, {getState, extra: {apiClient, validatePayload}}) => {
     const state = getState();
+    let req: CommerceFacetSearchRequest;
     validatePayload(facetId, requiredNonEmptyString);
-    const req = await buildCommerceFacetSearchRequest(
-      facetId,
-      state,
-      isFieldSuggestionsRequest
-    );
+    if (isRegularFacetSearchState(state, facetId)) {
+      req = await buildFacetSearchRequest(
+        facetId,
+        state,
+        isFieldSuggestionsRequest
+      );
+    } else {
+      req = await buildCategoryFacetSearchRequest(
+        facetId,
+        state,
+        isFieldSuggestionsRequest
+      );
+    }
```

Comment:
```suggestion
    const builder = isRegularFacetSearchState(state, facetId) ? buildFacetSearchRequest : buildCategoryFacetSearchRequest;
    const req = await builder(facetId, state, isFieldSuggestionsRequest);
```

Reply:
I can't do exactly that because the `state` argument has a different type in `buildFacetSearchRequest` and `buildCategoryFacetSearchRequest`, but I can do:

```
const req = isRegularFacetSearchState(state, facetId)
  ? buildFacetSearchRequest(facetId, state, isFieldSuggestionsRequest)
  : buildCategoryFacetSearchRequest(
      facetId,
      state,
      isFieldSuggestionsRequest
    );

const response = await apiClient.facetSearch(await req);
```